### PR TITLE
Add management command to send api move announcement email

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Part of #3742 by @sarayourfriend (does not fix because the issue also requires _sending_ the email, i.e., running the command in production).

## Description

Review the management command. The message text has already undergone review. The placeholder for the make post URL will be replaced before merging, after the post is available next week.

## Testing Instructions

Review the code and confirm tests are adequate and pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Upgrading this to high priority so that we can send it out early next week along with the make post on Monday. #2037 for context.

@WordPress/openverse-api pinging y'all for expedited review to match the priority.

File: api/api/management/commands/sendapimoveannouncement.py
Comment (dhruvkb) [inline api/api/management/commands/sendapimoveannouncement.py:25]:
```suggestion
Openverse only sends emails for critical operational notifications.
```

File: api/api/management/commands/sendapimoveannouncement.py
Comment (dhruvkb) [inline api/api/management/commands/sendapimoveannouncement.py:19]: This line should be wrapped like the others.

File: api/api/management/commands/sendapimoveannouncement.py
Comment (dhruvkb) [inline api/api/management/commands/sendapimoveannouncement.py:13]:
```suggestion
On 3 June 2024, the Openverse API’s domain will be changed to api.openverse.org and api.openverse.engineering will start
```

File: api/api/management/commands/sendapimoveannouncement.py
Comment (dhruvkb) [inline api/api/management/commands/sendapimoveannouncement.py:17]:
```suggestion
For the best experience, and to avoid redirects, please update code referencing api.openverse.engineering to use api.openverse.org.
```

**Review by stacimc — APPROVED** [review]
LGTM, tests are great. +1 to @dhruvkb's suggestions for the message text but approving to unblock.

I like the suggestion to make the command more general but strongly agree that should be a future iteration 👍

**Comment by sarayourfriend** [comment]
The issue references the old management command that we sent an email with before: https://github.com/WordPress/openverse/issues/3742

A general one is out of scope, as you've both said, but I want to share further context, as I don't believe we even need one. We've sent two emails in as many years, and the first was extremely specialised in the selection of who received it. It's hard to imagine it would be worth the time to write, test, and maintain a generalised management command when this one can be copy/pasted (and crucially in the meantime deleted, and not kept as unused code for who knows how long) and whatever modifications made as needed.

**Comment by dhruvkb** [comment]
Agree on both points: generalising is out of scope for this PR and this code can be deleted when unused (at 2 emails in ~3 years this is basically 100% of the time).

**Comment by sarayourfriend** [comment]
Because application emails are not unique, there is a chance of accidentally emailing people twice, if their multiple applications registered to a single email are pulled; that is, there is a chance of this if the `distinct()` call on the queryset is not filtering distinct on the application email. I need to check this. I'll do so by adding a new unit test first to determine whether this is the case, and if so, fix the `distinct` call or modify the loop to prevent it. I'll do that before merging this, of course, as well as addressing the editorial remarks from @dhruvkb.

@dhruvkb Just an FYI that I did ask for editorial reviews from everyone in the maintainer group before marking this PR ready for review; ideally these editorial remarks would have been shared there, as some of them overlap with content in the Make post. Just giving context for where the text came from, and the process it's already been through.

**Comment by sarayourfriend** [comment]
The emails are sent as of 2024-05-08T00:59:58.366Z.

**Pull request merged**
